### ISSUE TITLE
Cannot install cloudstack-marvin 4.15.0.0 on Ubuntu 20.04

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#4508]
see output below

```
root@mgt2004:~# apt install cloudstack-marvin
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 cloudstack-marvin : Depends: python-pip but it is not installable
E: Unable to correct problems, you have held broken packages.
```

it is because that there is no python2 pip on ubuntu 20.04 any more.
hence marvin cannot be installed, and integration test cannot be handled.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Marvin
~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### OS / ENVIRONMENT

Ubuntu 20.04

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by DaanHoogland** [comment]
@weizhouapache marvin 4.14 can be installed?

**Comment by weizhouapache** [comment]
> marvin 4.14 can be installed?

@DaanHoogland nee

**Comment by DaanHoogland** [comment]
> > marvin 4.14 can be installed?
> 
> @DaanHoogland nee, but I think we do not support ubuntu 20.04 in 4.14 so it is not an issue

right, but marvin on a operator machine could have been on 20.04 , right independent of the ACS install.

probably irrelevant, I think python2.7 end of live is likely to be the root cause here?!?
I encountered some instructions to start using `python -m pip` instead of straight `pip` to smoothen transition.
I'll install a ubuntu 20 to do some tinkering.

**Comment by weizhouapache** [comment]
> > > marvin 4.14 can be installed?
> > 
> > 
> > @DaanHoogland nee, but I think we do not support ubuntu 20.04 in 4.14 so it is not an issue
> 
> right, but marvin on a operator machine could have been on 20.04 , right independent of the ACS install.
> 
> probably irrelevant, I think python2.7 end of live is likely to be the root cause here?!?

@DaanHoogland yes exactly.
> I encountered some instructions to start using `python -m pip` instead of straight `pip` to smoothen transition.
> I'll install a ubuntu 20 to do some tinkering.
python/pip is replaced with python3/pip3 in ubuntu 20.04, issues in installation can be fixed by changes in debian/control and debian/cloudstack-marvin.postinst.

however, even if cloudstack-marvin is installed, there is still no way to use marvin to deploy zones or run nosetests, as our marvin python files support only python2 but there is no pip for python2 on ubuntu 20.04 .

in short, we need to change marvin to support python3.

**Comment by DaanHoogland** [comment]
right, I am doing that , but in combination with nose to nose2 upgrade in a poc. I'll give a seperation of the efforts a go.

**Comment by DaanHoogland** [comment]
btw @weizhouapache I think we can call this a major but not a critical, do you agree?

**Comment by weizhouapache** [comment]
> btw @weizhouapache I think we can call this a major but not a critical, do you agree?

@DaanHoogland yes, at least users should be able to install cloudstack-marvin on ubuntu 20.04

**Comment by rhtyd** [comment]
I think most users don't install marvin, this is only for test env to make it easier for a CI such as Trilian to do branch-based tests+marvin. As this is not a critical/blocker issue, let's handle this in 4.15.1.0 or master+.

**Comment by DaanHoogland** [comment]
@weizhouapache I have 4543 open for python3 (it still has about 14/86 failures in smoketests but at least marvin install. Can you test that on Ubuntu (targetted for 4.16)

**Comment by rhtyd** [comment]
Is it fixed now @weizhouapache, or maybe @DaanHoogland in your py3/marvin3 PR we can fix it? (just to make it installable that's all, make it run/pass would be more work)

**Comment by rhtyd** [comment]
@weizhouapache any update? Thanks.

**Comment by weizhouapache** [comment]
> @weizhouapache any update? Thanks.

@rhtyd, I will test it when @DaanHoogland finish #4727

**Comment by rhtyd** [comment]
@weizhouapache do you want to move milestone for this issue to 4.16? As #4727 is for master/4.16 and not 4.15.1.

**Comment by weizhouapache** [comment]
> @weizhouapache do you want to move milestone for this issue to 4.16? As #4727 is for master/4.16 and not 4.15.1.

@rhtyd yes, done

**Comment by nvazquez** [comment]
Hi @weizhouapache, PR #4727 has been merged, can you pleasse verify if it fixes the issue?

**Comment by weizhouapache** [comment]
@nvazquez great. it works.
